api.js

```diff
@@ -0,0 +1,164 @@
+"use strict";
+
+const {classes: Cc, interfaces: Ci, results: Cr, utils: Cu} = Components;
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+Cu.import("resource://devtools/shared/event-emitter.js");
+Cu.import("resource://gre/modules/ExtensionUtils.jsm");
+const {SingletonEventManager} = ExtensionUtils;
+
+XPCOMUtils.defineLazyModuleGetter(this, "webrtcUI",
+                                  "resource:///modules/webrtcUI.jsm");
+
+let origReceiveMessage = null;
```

Comment:
Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1310355 and moved the blocking callback handling over there.